### PR TITLE
Update locale pages for Adventure

### PR DESCRIPTION
**Pull request by mdcfe**
This PR rewrites the locale docs to better explain custom messages, as well as documenting the custom text command feature and upcoming chat component changes in 2.21.x.

### CODE REVIEW

File: pages/wiki/Locale.md [hunk 1]
```diff
@@ -1,109 +1,138 @@
...
 ## Customising messages
 
-Since all of the plugins' messages are defined in the messages file, you can fully customise the look and feel of your server by changing the text and colour scheme for all EssentialsX commands.
+In addition to using EssentialsX's built-in translations, you can also customise the text, appearance and functionality of EssentialsX messages!
 
-Simply place your edited locale file (with appropriate locale suffix) in the `plugins/Essentials` folder and make sure the locale setting in the config file matches (`locale: en` -> `messages_en.properties`), then restart the server.
+### Changing the default colours
 
-If you are having trouble, [try this tutorial video](https://www.youtube.com/watch?v=9Uan2oJ9zK8). Note that this video is for the original Essentials plugin, not EssentialsX.
+**NEW IN 2.21.0:** If you just want to change the colours used in the standard EssentialsX messages, you can do so using the `message-colors` section of `config.yml`:
```
Comment (JRoy) [inline pages/wiki/Locale.md:87]: 🦅
```suggestion
**NEW:** As of EssentialsX 2.21.0, you can change the colors used in the standard EssentialsX messages, just by changing the `message-colors` section of `config.yml`:
```

File: pages/wiki/Locale.md [hunk 2]
```diff
@@ -1,109 +1,138 @@
...
 
-You may need to update the file on occasion to deal with changes in EssentialsX such as parameter changes or changes to the actual syntax of the messages. This can be done with a diff tool like [Diffchecker](https://www.diffchecker.com/), which highlights changes between two text files.
+```yaml
+# Change the default primary and secondary colours used in EssentialsX messages.
+# Some messages may use custom colours, which will need to be edited in the appropriate message files.
```
Comment (JRoy) [inline pages/wiki/Locale.md:91]: 🦅
```suggestion
# Change the default primary and secondary colors used in EssentialsX messages.
# Some messages may use custom colors, which will need to be edited in the appropriate message files.
```

File: pages/wiki/Locale.md [hunk 3]
```diff
@@ -1,109 +1,138 @@
...
 
-When EssentialsX looks for a message, if it is missing from your version, it will instead use the **built-in version**.
-This means that if you only want to change a few messages instead of the whole translation, you can do this by making a copy of the locale file you're using in `plugins/Essentials/` and deleting all messages in the file, other than the ones you want to change. This means you don't have to worry about keeping your messages file up to date with every EssentialsX update.
+If you want to make more advanced changes to messages, you can create custom message files under the `plugins/Essentials` folder.  
+In this directory, create an empty file named `messages_<lang>.properties`, where `<lang>` is the language code for the language you want to edit.  
+You can now insert any customised messages in this file, and they will be loaded by EssentialsX the next time you restart your server. For example:
 
-When editing only a few keys, the process should look something like [this (click to open image)](http://wiki.ess3.net/w/images/0/0c/Locale.PNG).
+```properties
+TODOInsertExampleHere=<gold>aaaaaaaaa
```
Comment (JRoy) [inline pages/wiki/Locale.md:105]:
```suggestion
broadcast=<primary>Broadcast<secondary>>><white> {0}
```

File: pages/wiki/Locale.md [hunk 4]
```diff
@@ -1,109 +1,138 @@
...
+You can search through the default messages files on [GitHub](https://github.com/EssentialsX/Essentials/tree/2.x/Essentials/src/main/resources). We strongly recommend copying only the messages you want to edit, as this will make it easier to manage and update these files across EssentialsX versions.
 
-To use RGB colour codes, you will need to use Spigot's `§x§R§R§G§G§B§B` format, where RRGGBB come from the colour's hexadecimal code. You can also use [legacy colour codes (`§0` to `§f`)](https://minecraft.gamepedia.com/Formatting_codes#Color_codes) and [formatting codes](https://minecraft.gamepedia.com/Formatting_codes#Formatting_codes).
+**Note: do not edit any messages files directly inside the EssentialsX jar. You will lose these changes when you update the plugin and they will not be converted to the new MiniMessage format automatically.**
 
-## Example
+### Using rich text elements
 
-Let's say you simply want to change the way currency is output so that the currency symbol comes AFTER the balance:
+**NEW IN 2.21.0:** EssentialsX now supports the MiniMessage syntax for its messages. This allows you to include gradients, hoverable and clickable elements, custom fonts and more!
```
Comment (JRoy) [inline pages/wiki/Locale.md:116]:
```suggestion
**NEW:** As of 2.21.0, EssentialsX now supports the MiniMessage syntax for its messages. This allows you to include gradients, hoverable and clickable elements, custom fonts, and more!
```

**Review by JRoy — CHANGES_REQUESTED** [review]

File: pages/wiki/Locale.md [hunk 5]
```diff
@@ -1,109 +1,138 @@
...
 
-EssentialsX stores all its messages as templates in the messages file, and comes with support for [many languages already built-in](#Supported-languages). These files can be easily replaced or extended to customize your server by placing them in the `plugins/Essentials/` folder.
+EssentialsX includes messages for a wide range of languages out-of-the-box, as well as extensive support for customising almost every message in the plugin.
 
-## Switching language
+## Changing your server's language
 
-You can tell EssentialsX which language you want it to use by changing the `locale` setting in `config.yml`:
+By default, EssentialsX will attempt to use your system language if it is supported by the plugin.  
+You can change the default language by uncommenting and setting the `locale` option in `config.yml`:
```
[the remark this answers is not in the document]
Comment (mdcfe) [reply pages/wiki/Locale.md:8]: This is worded worse

File: pages/wiki/Locale.md [hunk 6]
```diff
@@ -1,109 +1,138 @@
-# Translations
+# Translations and Custom Messages
 
-EssentialsX stores all its messages as templates in the messages file, and comes with support for [many languages already built-in](#Supported-languages). These files can be easily replaced or extended to customize your server by placing them in the `plugins/Essentials/` folder.
+EssentialsX includes messages for a wide range of languages out-of-the-box, as well as extensive support for customising almost every message in the plugin.
```
[the remark this answers is not in the document]
Comment (mdcfe) [reply pages/wiki/Locale.md:3]: All user facing messages will be translatable but debug messages and stack traces aren't guaranteed to be translated

File: pages/wiki/Locale.md (hunk 2 above)
Comment (mdcfe) [reply pages/wiki/Locale.md:92]: This is the same wording as in the config

File: pages/wiki/Locale.md (hunk 3 above)
Comment (mdcfe) [reply pages/wiki/Locale.md]: Already went with another example

**Review by JRoy — DISMISSED** [review]

**Pull request merged**